integr8ly/integreatly-operator
# MGDAPI-6064 bump marin3r to 0.12.3

# Issue link
https://issues.redhat.com/browse/MGDAPI-6064

# What
Bump marin3r to 0.12.3

# Verification steps
- Fresh install will be verified as part of e2e on PR

- Upgrade scenario:

- Provision OSD cluster
- Create 1.38.0 catalog source:
```
cat << EOF | oc create -f -
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: managed-api-service
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/mstoklus/managed-api-service-index:1.38.0
EOF
```
- Run `make cluster/prepare/local`
- Create RHMI CR with: `IN_PROW=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml`
- Navigate to operator hub - find RHOAM and install under redhat-rhoam-operator ns
- Once fully installed change the catalog source `managed-api-service` under openshift-marketplace namespace value of index image from `:1.38.0` to `:1.39.0`
- After a minute or two, navigate to Installed Operators under redhat-rhoam-operator namespace and approve RHOAM 1.39.0 installation
- After few minutes upgrade should complete
- RHMI Cr should now show version 1.39.0 and Marin3r operator version of 0.12.3
- Navigate to 3scale namespace > deployments > select marin3r deployment > logs and confirm no obvious errors are there
- Navigate to cluster CRDs and search for envoyConfig CRD, open it up and double check that the annotation `operatorframework.io/installed-alongside-2a8b88c4bb8cc74e: redhat-rhoam-marin3r-operator/marin3r.v0.12.3` is present
- Navigate to 3scale namespace routes - open up the 3scale admin route
- Fetch admin password from system-seed secret and follow 3scale wizard to create sample product
- Navigate to redhat-rhoam-operator ns and scale down rhoam operator deployment
- Nagivate to redhat-rhoam-marin3r ns > configmaps > ratelimit-config and change the `max_value` to 5 > save changes
- Navigate to deployments under marin3r ns and restart marin3r deployment
- In 3scale admin portal go to integrations section of created product and copy the stage url
- Ping 6 times, 5 requests should go through and 6th one should be refused
